internal/services/apimanagement/api_management_resource.go

```diff
@@ -544,6 +559,22 @@ func resourceApiManagementService() *pluginsdk.Resource {
 				},
 			},
 
+			"public_ip_address_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"public_network_access": {
```

Comment:
Could we change this to be a boolean and match the rest of the provider?
```suggestion
			"public_network_access_enabled": {
```